packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -388,18 +474,26 @@ abstract class Layer extends AbstractNode with DiagnosticableTreeMixin {
   Layer? _previousSibling;
 
   @override
-  void dropChild(AbstractNode child) {
+  void dropChild(Layer child) {
+    assert(!_debugMutationsLocked);
     if (!alwaysNeedsAddToScene) {
       markNeedsAddToScene();
     }
+    if (child._childrenWithCompositionCallbacks != 0) {
+      _updateSubtreeCompositionObserverCount(child._childrenWithCompositionCallbacks);
```

Comment:
isn't this missing a `-`? The count needs to be _reduced_ by `child._childrenWithCompositionCallbacks` right?

If there is no failing test for this case, let's add one! :)

Reply:
You are right. I thought I had a test covering this but did not. Added one.